src/inference.c

```diff
@@ -103,3 +119,79 @@ Onnx__TensorProto** inference(Onnx__ModelProto *model, Onnx__TensorProto **input
 
   return _outputs;
 }
+
+/* Called once the model is loaded. Resolved each operator to a its function
+and popualtes all structures with the inputs */
+operator__context** resolve_check_get_input_and_attr(
+                                     Onnx__ModelProto *model,
+                                     Onnx__TensorProto **inputs,
+                                     int nInputs)
+{
+  operator__context **all_op_context;
+  /* malloc space for model->graph->n_node
+  *  malloc also space for TensorProto or AttributeProto depending on the
+  *  operator ?
+  */
+  for (int nodeIdx = 0; nodeIdx < model->graph->n_node; nodeIdx++)
+  {
+    /* New idea prototyping, just a proof of concept */
+    if (!strcmp(model->graph->node[nodeIdx]->op_type, "Add")){
+      printf("Calling operator Add\n");
+      operator__onnx__add__input *i;
+      operator__onnx__add__output *o;
+      operator__onnx__add__attribute *a;
+
+      /* Find input 0 (which is A) and in mnist will have different
+      names depending on the node:
+      -Convolution28_Output_0
+      -Convolution110_Output_0
+      -Times212_Output_0
+      */
+
+      /* This wont work as it is. For a given node, their inputs come from
+      three different sources.
+      1)Existing initializers in the model: model->graph->initializer[i]->name
+      2)Provided input to the model: inputs[i]->name
```

Comment:
isn't this the same thing? all provided initializers are registered as input

Reply:
I guess we can say its the same thing, but just wanted to point out that they come from different sources.